### ISSUE TITLE
Attention layer

### ISSUE THREAD

**Issue by samuel-g-bates** [calico/basenji#184]
Hello,

first off thank you for making/maintaining this repository. I see that you have the enformer attention layer in layers.py. I was wondering how I might define an enformer style model in a basenji parameters json file.

Thanks,
Sam

**Comment by davek44** [comment]
I added some params here that get you pretty close, and a description of the ways in which this will differ. The params will also work well with our new repository https://github.com/calico/baskerville, which I recommend switching to.

**Comment by samuel-g-bates** [comment]
Thank you for the help!

**Comment by samuel-g-bates** [comment]
The parameters produce this error when training on a single dataset.

    ValueError: Dimensions must be equal, but are 1536 and 896 for '{{node poisson/mul}} = Mul[T=DT_FLOAT](y, poisson/Log)' with input shapes: [2,1536,1], [2,896,1].

Its from trainer.py at loss = self.loss_fn(y, pred) + sum(model.losses)

I believe it is caused by pred being cropped to 896 but y not being cropped.

Any guidance on how to fix this would be greatly appreciated.

Here is the code I run up to the error:
```
basenji_data.py  -l 196608 --local -o data -t .1 -v .1 -w 128 ce11_genome.fa data.txt
hound_train.py params_enformer.json data
```

**Comment by davek44** [comment]
Right, you need to either drop the Cropping layers from the params json file, or introduce cropping into your dataset construction using this option https://github.com/calico/basenji/blob/master/bin/basenji_data.py#L60

For what it's worth, I've done a lot of work on C elegans that we never ended up publishing. I never found Enformer's self-attention layers to work any better than Basenji's dilated convolution layers. I believe the smaller genome doesn't offer enough training data to support the more complex operation. The genes are also much more compact, so 196k is overkill. Reach out by email if you want more advice.
